src/stages/models/manager.ts

```diff
@@ -201,7 +202,7 @@ export class StageManager {
     const isValidStatus = updateActor.getSnapshot().can({ type: nextStatusChange });
 
     if (!isValidStatus) {
-      throw new InvalidUpdateError(commonErrorMessages.invalidStatusChange);
+      throw new IllegalStageStatusTransitionError(stagesErrorMessages.illegalStageStatusTransitionError);
```

Comment:
i would consider giving better messages like what the current and whats the requested status.